packages/tooltip/src/use-tooltip.ts

```diff
@@ -117,29 +117,30 @@ export function useTooltip(props: UseTooltipProps = {}) {
   const exitTimeout = React.useRef<number>()
 
   const openWithDelay = React.useCallback(() => {
-    if (!isDisabled) {
+    if (!isDisabled && !enterTimeout.current) {
       enterTimeout.current = window.setTimeout(onOpen, openDelay)
     }
   }, [isDisabled, onOpen, openDelay])
 
   const closeWithDelay = React.useCallback(() => {
     if (enterTimeout.current) {
       clearTimeout(enterTimeout.current)
+      enterTimeout.current = undefined
     }
     exitTimeout.current = window.setTimeout(onClose, closeDelay)
   }, [closeDelay, onClose])
 
   const onClick = React.useCallback(() => {
-    if (closeOnClick) {
+    if (isOpen && closeOnClick) {
```

Comment:
Avoid clearing an enter timer before opening the tooltip when clicking the trigger button